examples/02-javascript-basics/array_manipulation_library.js

```diff
@@ -11,93 +11,132 @@ Join - join the array elements into a string given a parameter
 */
 
 const $ = {
-	filter: function () {},
-	map: function () {},
-	reduce: function () {},
-	head: function () {},
-	tail: function () {},
-	join: function () {}
+  filter: function (arr, fx) {
+    res = []
+    for (i = 0; i < arr.length; i++) {
+      if (fx(arr[i]))
+        res.push(arr[i])
+    }
+    return res
+  },
+  map: function (arr, fx) {
+    res = []
+    for (i = 0; i < arr.length; i++) {
+        res.push(fx(arr[i]))
+    }
+    return res
+  },
+  reduce: function (arr, fx, initialValue) {
+    if (arr.length === 0) {
+      if (initialValue === undefined)
```

Comment:
Prefer to add brackets as JS sometimes have some weird execution happening without braces.